src/module/sheets/AbstractTwodsixActorSheet.ts

```diff
@@ -140,13 +141,19 @@ export abstract class AbstractTwodsixActorSheet extends ActorSheet {
       type,
       data
     };
-
     // Remove the type from the dataset since it's in the itemData.type prop.
     // delete itemData.data.type;
     this.updateWithItemSpecificValues(itemData, <string>type);
 
-    // Finally, create the item!
-    await this.actor.createEmbeddedDocuments("Item", [itemData]);
+    const items = <TwodsixItem[]>await this.actor.createEmbeddedDocuments("Item", [itemData]);
+
+    if (type === "trait") {
+      const effects = await this.actor.createEmbeddedDocuments("ActiveEffect", [{
+        origin: items[0].uuid,
+        icon: "systems/twodsix/assets/icons/science.svg"
```

Comment:
Maybe use the icon from the trait instead?  Most of the CD traits have icons.

Reply:
Good idea